packages/zoe/test/unitTests/contractSupport/test-zoeHelpers.js

```diff
@@ -564,6 +561,95 @@ test('ZoeHelpers canTradeWith allocation different than give', t => {
   }
 });
 
+test('ZoeHelpers canTradeWithMapKeywords', t => {
+  t.plan(2);
+  const { moolaR, simoleanR, moola, simoleans, amountMaths } = setup();
+  const leftOfferHandle = harden({});
+  const rightOfferHandle = harden({});
+  const cantTradeRightOfferHandle = harden({});
+  try {
+    const mockZCF = harden({
+      getInstanceRecord: () =>
+        harden({
+          issuerKeywordRecord: {
+            Asset: moolaR.issuer,
+            Price: simoleanR.issuer,
+          },
+          keywords: ['Asset', 'Price'],
+        }),
+      getAmountMath: brand => amountMaths.get(brand.getAllegedName()),
+      getCurrentAllocation: handle => {
+        switch (handle) {
+          case leftOfferHandle:
+            return harden({
+              Asset: moola(10),
+              Price: simoleans(4),
+            });
+          case rightOfferHandle:
+            return harden({
+              Price: simoleans(6),
+              Asset: moola(7),
+            });
+          case cantTradeRightOfferHandle:
+            return harden({
+              Price: simoleans(6),
+              Asset: moola(100),
+            });
+          default:
+            throw new Error('unexpected handle');
+        }
+      },
+      getZoeService: () => {},
+      getOffer: handle => {
+        if (handle === leftOfferHandle) {
+          return harden({
+            proposal: {
+              give: { Asset: moola(10) },
+              want: { Price: simoleans(4) },
+              exit: { onDemand: null },
+            },
+          });
+        }
+        if (handle === rightOfferHandle) {
+          return harden({
+            proposal: {
+              give: { Price: simoleans(6) },
+              want: { Asset: moola(7) },
+              exit: { onDemand: null },
+            },
+          });
+        }
+        if (handle === cantTradeRightOfferHandle) {
+          return harden({
+            proposal: {
+              give: { Price: simoleans(6) },
+              want: { Asset: moola(100) },
+              exit: { onDemand: null },
+            },
+          });
+        }
+        throw new Error('unexpected handle');
+      },
+    });
+    const { canTradeWithMapKeywords } = makeZoeHelpers(mockZCF);
+    t.ok(
+      canTradeWithMapKeywords(leftOfferHandle, rightOfferHandle, [
```

Comment:
Can we have a test with different keywords between the offers? It seems weird to test with the same ones.